patches/tModLoader/Terraria/PopupText.cs.patch

```diff
@@ -17,11 +34,29 @@
  				else if (newItem.rare == -11)
  					Main.popupText[num2].color = new Color(255, 175, 0);
  				else if (newItem.rare == -1)
-@@ -215,6 +_,7 @@
+@@ -211,10 +_,14 @@
+ 					Main.popupText[num2].color = new Color(5, 200, 255);
+ 				else if (newItem.rare == 10)
+ 					Main.popupText[num2].color = new Color(255, 40, 100);
+-				else if (newItem.rare >= 11)
++				else if (newItem.rare == 11)
  					Main.popupText[num2].color = new Color(180, 40, 255);
++				else if (newItem.rare > 11)
++					Main.popupText[num2].color = ModRarity.GetRarity(newItem.rare).RarityColor();
  
++				Main.popupText[num2].rarity = newItem.rare;
  				Main.popupText[num2].expert = newItem.expert;
 +				Main.popupText[num2].master = newItem.master;
  				Main.popupText[num2].name = newItem.AffixName();
  				Main.popupText[num2].stack = stack;
  				Main.popupText[num2].velocity.Y = -7f;
+@@ -398,6 +_,9 @@
+ 				color = new Color((byte)Main.DiscoR, (byte)Main.DiscoG, (byte)Main.DiscoB, Main.mouseTextColor);
+ 			else if (master)
+ 				color = new Color(255, (byte)(Main.masterColor * 200f), 0, Main.mouseTextColor);
++
++			if (rarity > ItemRarityID.Purple && ModRarity.GetRarity(rarity).Animated())
```

Comment:
Why the if check? Why not just set color to `ModRarity.GetRarity(rarity).RarityColor();` regardless, why does it need to be 'animated'? (you can create animations using the function itself)